src/connector/src/parser/mod.rs

```diff
@@ -754,9 +791,6 @@ impl ByteStreamSourceParserImpl {
         let protocol = &parser_config.specific.protocol_config;
         let encode = &parser_config.specific.encoding_config;
         match (protocol, encode) {
-            (ProtocolProperties::Plain, EncodingProperties::Json(_)) => {
-                JsonParser::new(parser_config.specific, rw_columns, source_ctx).map(Self::Json)
-            }
```

Comment:
here I remove the plain json parser as it is hard to ingest kafka key. This can have overhead because it uses a `[&str]` to fetch both key and payload fields.